civihr_employee_portal/src/View/CustomFieldsMappingFixer.php

```diff
@@ -0,0 +1,164 @@
+   *
+   * @var array $option
+   *     structure that contains an option for a field in view
+   */
+  private static function processOption(&$option) {
+    // checking if this table is part of the civicrm views integration
+    $tableName = self::getCiviCRMtableName($option['table']);
+    $prefix = $tableName ? self::getPrefix($tableName) : false;
+    if ( $prefix ) {
+      $option['table'] = $tableName;
+      $columns = self::getColumnsFromTable($prefix . $tableName);
+      // if the table where the current field belongs exists does not have it
+      // then we can search for similar fields in the same table and make replacements
+      if (!isset($columns[$option['field']])) {
+        $fieldNameBase = self::removeLastNumberOfString($option['field']);
+        $pos = array_search($fieldNameBase, $columns);
+        if ($pos !== FALSE) {
+          $option['field'] = array_search($columns[$pos], $columns);
+        }
+      }
+    }
+  }
+
+  /**
+   * Removes the number in end of a string
+   * In this context that number is always separated by an underscore
+   *
+   * @param  string $string
+   *     the string which could have underscores or not
+   *
+   * @return string
+   *     the string without the number at the end,
+   *     if no number found at the end returns the same string
+   */
+  private static function removeLastNumberOfString($string) {
+    $pieces = explode('_', $string);
+    $lastIndex = count($pieces) - 1;
+    return $lastIndex > 0 && intval(array_pop($pieces)) ? implode('_', $pieces) : $string;
+  }
+
+  /**
+   * Return an existing CiviCRM table based on the base name of the table,
+   * where "base name of the table" is the name of the table without the number
+   * at the end
+   *
+   * @param string $tableName
+   *     this is the table name exported by views
+   *
+   * @return string
+   *     the name of the existing table if found or an empty string
+   */
+  private static function getCiviCRMtableName($tableName) {
+    global $databases;
+    static $tableNamesMap = [];
+    // creates a map with table names without IDs to existing table names with IDs
+    if (!$tableNamesMap) {
+      $tablesWithIds = array_keys($databases['default']['default']['prefix']);
+      foreach ($tablesWithIds as $tableWithId) {
+        $tableNamesMap[self::removeLastNumberOfString($tableWithId)] = $tableWithId;
+      }
+    }
+
+    $exportedTableBaseName = self::removeLastNumberOfString($tableName);
+    $tableFound = isset($tableNamesMap[$exportedTableBaseName]);
+    return $tableFound ? $tableNamesMap[$exportedTableBaseName] : '';
+  }
+
+  /**
+   * To obtain the prefix for a table in the current installation
+   *
+   * @param string $tableName
+   *     the name of the table
+   *
+   * @return string
+   *     the prefix for the table
+   */
+  private static function getPrefix($tableName) {
+    global $databases;
+    return $databases['default']['default']['prefix'][$tableName];
```

Comment:
Please avoid using global variables if you can, it's better to use a class that allows access to this in case something changes with the underlying implementation later.

Reply:
@mickadoo I will search for code that returns this information and will use it in case I found it.